src/EditorFeatures/Core.Wpf/Lightup/LightupHelpers.cs

```diff
@@ -328,21 +329,26 @@ public static Action<T> CreateActionAccessor<T>(Type? type, string methodName)
             throw new InvalidOperationException($"Method '{method}' produces a value of type '{method.ReturnType}', which is not assignable to type '{typeof(TResult)}'");
         }
 
-        var parameter = Expression.Parameter(typeof(T), GenerateParameterName(typeof(T)));
-        var argument = Expression.Parameter(typeof(TArg), parameters[0].Name);
-        var instance =
+        var instanceParameter =
             type.GetTypeInfo().IsAssignableFrom(typeof(T).GetTypeInfo())
-            ? (Expression)parameter
-            : Expression.Convert(parameter, type);
+                ? Expression.Parameter(type, GenerateParameterName(type))
+                : Expression.Parameter(typeof(T), GenerateParameterName(typeof(T)));
```

Comment:
`parameter` should not change from its original definition

Reply:
Alright, get an exception at runtime, working on it now.

Reply:
OK done, need one more ConvertExpression

Reply:
`parameter` still has the wrong type here.